RELEASES.md

```diff
@@ -7,6 +7,14 @@ Releases are proposed and adopted or rejected using the usual [project governanc
 An anti-pattern that we want to avoid is heavy development or discussions "late cycle" around major releases.
 We want to build a community that is involved and communicates consistently through all releases instead of relying on "silent periods" as a judge of stability.
 
+## Release Process
```

Comment:
Non-maintainers have to understand neither, all maintainers have to understand the current `RELEASES.md`, and only @vbatts (or whoever is cutting the release) needs to understand the notes you're moving here.